test/index.js

```diff
@@ -551,34 +665,75 @@ describe('def', function() {
-                   '{ x :: Number, y :: Number } ' +
-                   'as its first argument; received {}'));
+                   'Invalid value\n' +
+                   '\n' +
+                   'dist :: { x :: Number, y :: Number } -> { x :: ' +
+                   'Number, y :: Number } -> Number\n' +
+                   '        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
+                   '                     1\n' +
+                   '\n' +
+                   '1)  {} :: Object\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of { x :: ' +
+                   'Number, y :: Number }.\n'));
 
     throws(function() { dist({x: 0}); },
            errorEq(TypeError,
-                   '‘dist’ expected a value of type ' +
-                   '{ x :: Number, y :: Number } ' +
-                   'as its first argument; received {"x": 0}'));
+                   'Invalid value\n' +
+                   '\n' +
+                   'dist :: { x :: Number, y :: Number } -> { x :: ' +
+                   'Number, y :: Number } -> Number\n' +
+                   '        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
+                   '                     1\n' +
+                   '\n' +
+                   '1)  {"x": 0} :: Object\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of { x :: ' +
+                   'Number, y :: Number }.\n'));
 
     throws(function() { dist({x: 0, y: null}); },
            errorEq(TypeError,
-                   '‘dist’ expected a value of type ' +
-                   '{ x :: Number, y :: Number } ' +
-                   'as its first argument; received {"x": 0, "y": null}'));
+                   'Invalid value\n' +
+                   '\n' +
+                   'dist :: { x :: Number, y :: Number } -> { x :: ' +
+                   'Number, y :: Number } -> Number\n' +
+                   '        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
+                   '                     1\n' +
+                   '\n' +
+                   '1)  {"x": 0, "y": null} :: Object\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of { x :: ' +
+                   'Number, y :: Number }.\n'));
 
     throws(function() { length({start: 0, end: 0}); },
            errorEq(TypeError,
-                   '‘length’ expected a value of type ' +
-                   '{ end :: { x :: Number, y :: Number }' +
-                   ', start :: { x :: Number, y :: Number } } ' +
-                   'as its first argument; received {"end": 0, "start": 0}'));
+                   'Invalid value\n' +
+                   '\n' +
+                   'length :: { end :: { x :: Number, y :: Number }, start ' +
+                   ':: { x :: Number, y :: Number } } -> Number\n' +
+                   '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^' +
+                   '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
```

Comment:
At first glance I thought we were one caret short. Let's make this tiny change:

```diff
                    'length :: { end :: { x :: Number, y :: Number }, start ' +
                    ':: { x :: Number, y :: Number } } -> Number\n' +
-                   '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^' +
-                   '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
+                   '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^' +
+                   '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
```

By moving one caret to the second line the last caret lines up with the closing squiggly bracket.